coral/src/app/features/components/table-filters/OperationTypeFilter.test.tsx

```diff
@@ -0,0 +1,164 @@
+import { cleanup, screen, waitFor, within } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import {
+  operationTypeList,
+  requestOperationTypeNameMap,
+} from "src/app/features/approvals/utils/request-operation-type-helper";
+import { OperationTypeFilter } from "src/app/features/components/table-filters/OperationTypeFilter";
+import { customRender } from "src/services/test-utils/render-with-wrappers";
+
+const filterLabel = "Filter by operation type";
+
+describe("OperationTypeFilter.tsx", () => {
+  describe("renders all necessary elements", () => {
+    beforeAll(async () => {
+      customRender(<OperationTypeFilter />, {
+        memoryRouter: true,
+      });
+    });
+
+    afterAll(cleanup);
+
+    it("shows a select element for operation type", () => {
+      const select = screen.getByRole("combobox", {
+        name: filterLabel,
+      });
+
+      expect(select).toBeEnabled();
+    });
+
+    it("renders a list of options for operation types", () => {
+      const allOption = screen.getByRole("option", {
+        name: "All operation types",
+      });
+      expect(allOption).toBeEnabled();
+
+      operationTypeList.forEach((type) => {
+        const option = screen.getByRole("option", {
+          name: requestOperationTypeNameMap[type],
+        });
+
+        expect(option).toBeEnabled();
+      });
+    });
+  });
+
+  describe("sets the active operation type based on a query param", () => {
+    const deleteOperation = "DELETE";
+    const deleteName = requestOperationTypeNameMap[deleteOperation];
+
+    beforeEach(async () => {
+      const routePath = `/?operationType=${deleteOperation}`;
+
+      customRender(<OperationTypeFilter />, {
+        memoryRouter: true,
+        queryClient: true,
+        customRoutePath: routePath,
+      });
+    });
+
+    afterEach(() => {
+      cleanup();
+    });
+
+    it(`shows DELETE name as the active option one`, async () => {
+      const select = await screen.findByRole("combobox", { name: filterLabel });
+      const option = await within(select).findByRole("option", {
+        name: deleteName,
+        selected: true,
+      });
+
+      expect(select).toHaveValue(deleteOperation);
+      expect(option).toBeVisible();
+      expect(option).toHaveValue(deleteOperation);
+    });
+  });
+
+  describe("handles user selecting a environment", () => {
```

Comment:
`environment` -> `operation type`? (in some other tests as well)

Reply:
Yikes, there were so many. Copy and pasting tests is nice, but also not nice.

[4b7e314](https://github.com/aiven/klaw/pull/888/commits/4b7e314f99a922f16c1c19de2a1fe72459f2389c)